AutoGrade/grader.py

```diff
@@ -80,23 +63,14 @@ def run_student_tests(target_folder, total_points, timeout):
 
     score = (0, 0) # passed, failed
 
-    logger.debug("Capturing stdout")
-
     out_file =  os.path.join(target_folder, "test-results" + ".log")
     touch(out_file)
 
-    p = Process(target=run_test, args=(os.path.basename(out_file), target_folder, timeout,))
-    logger.debug("Starting test process for submission")
-    p.start()
-    p.join(timeout + 1) # Pytest will also timeout
-
-    #In case process is stuck in infinite loop or something
-    if p.is_alive():
-        logger.debug("Terminating process [TIMEOUT]")
-        p.terminate()
-        with open(out_file, 'w') as f:
-            f.write("\n\nProcess Terminated due to timeout.")
-
+    with open(out_file, 'w') as f:
+        process = Popen(["py.test", "--timeout", str(timeout)], cwd=target_folder, stdout=f, stderr=f, shell=True)
```

Comment:
@recluze Using Popen to run the py.test

Reply:
In my PC pytest is accessed by `py.test`. Checked it on Ubuntu (Bash on Ubuntu) and `py.test` is also working there.

Reply:
Hnn, let's first figure out this behavior discrepency before we commit. Don't want to have to handle issues of 100+ students on prod. 

Version differences? Here's what I'm using on prod: 

```
platform linux2 -- Python 2.7.10, pytest-3.2.1, py-1.4.34, pluggy-0.4.0
rootdir: /home/sys-autograder/autograder, inifile:
plugins: timeout-1.2.0
``` 

And here's my machine: 

```
platform darwin -- Python 2.7.13, pytest-3.0.7, py-1.4.33, pluggy-0.4.0
rootdir: /Users/nam/temp/as-test, inifile:
plugins: timeout-1.2.0
```

I verified on prod and it also produces the desired behvior. 

Can you please see here: https://pypi.python.org/pypi/pytest-timeout ... and see if another "timeout method" works for your machine?

Reply:
py.test on my machine
```
platform win32 -- Python 2.7.12, pytest-3.2.3, py-1.4.34, pluggy-0.4.0
rootdir: C:\Users\SyedOwais\Desktop\assignment1, inifile:
plugins: timeout-1.2.0```

py.test on "Ubuntu"
```
platform linux2 -- Python 2.7.12, pytest-3.2.3, py-1.4.34, pluggy-0.4.0
rootdir: /mnt/c/Users/SyedOwais/Desktop/assignment1, inifile:
plugins: timeout-1.2.0
```
Here is the complete output of pytest:
https://gist.github.com/soachishti/41e1c0f2a04de2f9ca07d5f936339592

And getting same result for different methods.